### PR TITLE
AAA_Install rose suite.conf in cylc rose

### LINKED ISSUE #36: Install rose-suite.conf in cylc rose
**Describe exactly what you would like to see in an upcoming release**

Cylc-rose and _not_ Cylc-flow should install the ``rose-suite.conf`` file from the source directory rather than the work directory. This will allow more complex behavior from the plugin in future, such as diffing the old and new ``rose-suite.conf``

**Additional context**

You will need to add an exclusion in cylc-flow/cylc/flow/suite_files.py as a companion to changes to cylc-rose.

**Pull requests welcome!**

### PR DESCRIPTION
**Pull request by wxtim**
Closes #36, when run in combination with [cylc-flow PR 4094](https://github.com/cylc/cylc-flow/pull/4094)

### CODE REVIEW

File: tests/functional/test_copy_config_file.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            True
+        ),
+        (
+            # Valid sourcedir with rose file, rose file at dest:
+            {
+                'src/rose-suite.conf': '[env]\nFOO=2',
+                'dest/rose-suite.conf': '[env]\nFOO=1'
+            },
+            {'dir_': 'src', 'dest_root': 'dest'},
+            True
```
Comment (oliver-sanders) [inline tests/functional/test_copy_config_file.py:64]: I think this is the only test we need here, just to make sure that the new contents are copied over the old.

**Review by oliver-sanders — APPROVED** [review]

File: tests/functional/test_copy_config_file.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        ),
+        (
+            # Valid sourcedir with rose file, no rose file at dest:
+            {
+                'src/rose-suite.conf': '[env]\nFOO=1',
+                'dest/jelly.txt': 'any-old-stuff'
+            },
+            {'dir_': 'src', 'dest_root': 'dest'},
+            True
+        ),
```
Comment (wxtim) [inline tests/functional/test_copy_config_file.py:56]:
```suggestion
```

File: tests/functional/test_copy_config_file.py (hunk 1 above)
Comment (wxtim) [reply tests/functional/test_copy_config_file.py:37]: Killed the rest.

File: cylc/rose/entry_points.py [hunk 3]
```diff
@@ -217,3 +219,28 @@ def record_cylc_install_options(
     )
     dumper(rose_suite_conf, rose_conf_filepath)
     return cli_config, rose_suite_conf
+
+
+def copy_config_file(
+    dest_root=None,
+    opts=None,
+    dir_=None,
+):
```
Comment (MetRonnie) [inline cylc/rose/entry_points.py:228]: Docstring 😋

File: cylc/rose/entry_points.py [hunk 4]
```diff
@@ -217,3 +219,28 @@ def record_cylc_install_options(
...
+        dir_ is None
+    ):
+        return False
+
+    rundir = Path(dest_root)
+    srcdir = Path(dir_)
+    srcdir_rose_conf = srcdir / 'rose-suite.conf'
+    rundir_rose_conf = rundir / 'rose-suite.conf'
+
+    if not srcdir_rose_conf.is_file():
```
Comment (MetRonnie) [inline cylc/rose/entry_points.py:240]: One thing to bear in mind, `.is_file()` returns `True` if it's a symlink. I guess it's probably pretty unlikely that someone would make `rose-suite.conf` a symlink, but possible?

Comment (wxtim) [reply cylc/rose/entry_points.py:240]: If it is then we end up copying the target using `shutil.copy2(srcdir_rose_conf, rundir_rose_conf)`, which seems to me like reasonable behaviour.

File: cylc/rose/entry_points.py [hunk 5]
```diff
@@ -226,6 +226,20 @@ def copy_config_file(
...
 ):
+    """Copy the ``rose-suite.conf`` from a workflow source to run directory.
+
+    Args:
+        dir_ (pathlib.Path | or str):
+            Source Path of Cylc install.
+        opts:
+            Not used in this function, but requried for consistent entry point.            
+        dest_root (pathlib.Path | or str):
+            Destination path of Cylc install - the workflow rundir.
```
Comment (MetRonnie) [inline cylc/rose/entry_points.py:237]: Thanks, but order is backwards

File: cylc/rose/entry_points.py [hunk 6]
```diff
@@ -217,3 +219,42 @@ def record_cylc_install_options(
     )
     dumper(rose_suite_conf, rose_conf_filepath)
     return cli_config, rose_suite_conf
+
+
+def copy_config_file(
+    dest_root=None,
+    opts=None,
+    dir_=None,
+):
```
Comment (wxtim) [inline cylc/rose/entry_points.py:228]:
```suggestion
def copy_config_file(
    dir_=None,
    opts=None,
    dest_root=None,
):
```

File: cylc/rose/entry_points.py (hunk 5 above)
Comment (wxtim) [reply cylc/rose/entry_points.py:237]: Fixed, but by changing the order of the kwargs. Bear in mind that I'll be refactoring this early next week. I think that the order should be "sourcedir", "opts", "rundir".

**Review by MetRonnie — APPROVED** [review]

**Pull request merged**